src/test/java/io/cryostat/net/web/http/api/beta/TargetRecordingMetadataPatchHandlerTest.java

```diff
@@ -174,100 +179,98 @@ void shouldUpdateLabels() throws Exception {
+                    .thenReturn(descriptor);
+            Mockito.when(descriptor.isPresent()).thenReturn(true);
 
             Mockito.when(targetConnectionManager.executeConnectedTask(Mockito.any(), Mockito.any()))
                     .thenAnswer(
-                            new Answer<Object>() {
-                                @Override
-                                public Object answer(InvocationOnMock invocation) throws Throwable {
-                                    TargetConnectionManager.ConnectedTask task =
-                                            invocation.getArgument(1);
-                                    return task.execute(connection);
-                                }
-                            });
-
-            Mockito.when(recordingTargetHelper.getDescriptorByName(connection, recordingName))
-                    .thenReturn(Optional.of(descriptor));
+                            arg0 ->
+                                    ((TargetConnectionManager.ConnectedTask<Object>)
+                                                    arg0.getArgument(1))
+                                            .execute(connection));
 
+            CompletableFuture<String> labelFuture = Mockito.mock(CompletableFuture.class);
+            labelFuture.complete(labels);
             Mockito.when(
-                            resp.putHeader(
-                                    Mockito.any(CharSequence.class),
-                                    Mockito.any(CharSequence.class)))
-                    .thenReturn(resp);
+                            recordingMetadataManager.addRecordingLabels(
+                                    targetId, recordingName, labels))
+                    .thenReturn(labelFuture);
 
-            handler.handle(ctx);
+            IntermediateResponse<String> response = handler.handle(requestParameters);
+            MatcherAssert.assertThat(response.getStatusCode(), Matchers.equalTo(200));
 
-            Mockito.verify(
-                    recordingMetadataManager.addRecordingLabels(targetId, recordingName, labels));
             Mockito.verify(notificationFactory).createBuilder();
             Mockito.verify(notificationBuilder).metaCategory("RecordingMetadataUpdated");
             Mockito.verify(notificationBuilder).metaType(HttpMimeType.JSON);
             Mockito.verify(notificationBuilder)
-                    .message(Map.of("recording", recordingName, "target", targetId));
+                    .message(
+                            Map.of(
+                                    "target",
+                                    targetId,
+                                    "recording",
+                                    recordingName,
+                                    "labels",
+                                    labels));
             Mockito.verify(notificationBuilder).build();
             Mockito.verify(notification).send();
```

Comment:
Before I replace the internal `labels` strings with maps, I wanted to update the unit tests to make sure everything else works properly. I'm getting an error with the `notificationBuilder` here. The notification stubs in `setup()` are the same as other tests. What am I missing?
```
Expected [notificationBuilder.message(

    {"recording" = "someRecording", "target" = "fooTarget", "labels" = "{key=value}"}

);] but was [notificationBuilder.metaCategory(

    "RecordingMetadataUpdated"

);

notificationBuilder.metaType(

    JSON

);

notificationBuilder.message(

    {"recordingName" = "someRecording", "target" = "fooTarget", "labels" = "{key=value}"}

);

notificationBuilder.build(

    

);]
```

Reply:
That looks a little funny. I wonder if it's something silly like the `labels` in one is actually a string literal, but in the other it's some other expected object reference value that gets formatted the same way by its `toString()` to print out that error message. A Map's `toString()` might look like that for example, but the Map is not actually equal to the string.

Reply:
Ah whoops, I changed the message "recording" to "recordingName" - thanks for looking at this anyway